app/src/main/java/com/example/habitsmasher/UserValidator.java

```diff
@@ -3,9 +3,7 @@
 import android.util.Patterns;
 import android.widget.EditText;
 
-import androidx.core.util.PatternsCompat;
-
-public class UserValidator {
+public class UserValidator extends EmailValidator{
```

Comment:
Hmm... Let's not **extend** `EmailValidator` here. All I meant by "use the email validator in the user validator" was that let's use the email validation methods that we created in the email validator, in the user validator.

This would mean we can create **a new instance** of `EmailValidator` inside the UserValidator, and use the email verification methods. No need to tie these classes together using inheritance.

- [ ] create an instance of `EmailValidator` inside `UserValidator` and replace any existing email validation with calls to methods in the email validator
- [ ] undo the **extending** of the `EmailValidator` class